newIDE/app/src/SceneEditor/CustomObjectExtractor/CustomObjectExtractor.spec.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import { extractAsCustomObject } from './CustomObjectExtractor';
+import { makeTestProject } from '../../fixtures/TestProject';
+import Rectangle from '../../Utils/Rectangle';
+import { getInstancesInLayoutForObject } from '../../Utils/Layout';
+
+const gd: libGDevelop = global.gd;
+
+describe('CustomObjectExtractor', () => {
+  it('can extract 2 instances as a custom object', () => {
+    const { project, testLayout } = makeTestProject(gd);
+
+    const initialInstances = testLayout.getInitialInstances();
+    const leftInstance = initialInstances.insertNewInitialInstance();
+    leftInstance.setObjectName('MySpriteObject');
+    leftInstance.setX(200);
+    leftInstance.setY(500);
+    leftInstance.setCustomWidth(100);
+    leftInstance.setCustomHeight(150);
+    const rightInstance = initialInstances.insertNewInitialInstance();
+    rightInstance.setObjectName('MySpriteObject');
+    rightInstance.setX(300);
+    rightInstance.setY(500);
+    rightInstance.setCustomWidth(100);
+    rightInstance.setCustomHeight(120);
+    const selectedInstances = [leftInstance, rightInstance];
+
+    const deleteSelection = () => {
+      initialInstances.removeInstance(leftInstance);
+      initialInstances.removeInstance(rightInstance);
+    };
+
+    extractAsCustomObject(
+      project,
```

Comment:
I'm not sure it's necessary because it's not called with plain values. It allows to chose a parameter names that make more sense for the implementation without having to specify them here: 
```JS
    extractAsCustomObject(
      project,
      globalObjectsContainer,
      objectsContainer,
      initialInstances,
      chosenExtensionName,
      isNewExtension,
      chosenEventsBasedObjectName,
      shouldRemoveSceneObjectsWhenNoMoreInstance,
      instancesSelection.getSelectedInstances(),
      editorDisplay.instancesHandlers.getSelectionAABB(),
      deleteSelection,
      onExtractAsEventBasedObject
    );
```

Reply:
```JS
    extractAsCustomObject({
      project,
      globalObjectsContainer,
      objectsContainer,
      initialInstances,
      chosenExtensionName,
      isNewExtension,
      chosenEventsBasedObjectName,
      shouldRemoveSceneObjectsWhenNoMoreInstance,
      instancesSelection.getSelectedInstances(),
      editorDisplay.instancesHandlers.getSelectionAABB(),
      deleteSelection,
      onExtractAsEventBasedObject
    });
```

would work as well and will save us when a function () => void is replaced by another () => void because an argument was added but was mismatched during a search/replace?